tests/fixtures/model_fixture.py

```diff
@@ -32,30 +35,27 @@ def upload_model(bucketfs_location: AbstractBucketFSLocation,
 @contextmanager
 def upload_model_to_local_bucketfs(
         model_name: str, download_tmpdir: Path) -> str:
-
-    download_model(model_name, download_tmpdir)
-    upload_tmpdir_name = Path(download_tmpdir, "upload_tmpdir")
-    upload_tmpdir_name.mkdir(parents=True, exist_ok=True)
+    upload_tmpdir_name = download_model(model_name, download_tmpdir)
     bucketfs_location = LocalFSMockBucketFSLocation(
         PurePosixPath(upload_tmpdir_name))
-    upload_model(bucketfs_location, model_name, download_tmpdir)
-    yield upload_tmpdir_name
+    upload_model(bucketfs_location, model_name, upload_tmpdir_name)
+    yield download_tmpdir
```

Comment:
this looks strange

Reply:
changed upload_tempdir_name to upload_tempdir. if this is not what you meant please tell me.

Reply:
no, I mean that we yield download_tmpdir is confusing me, I would yield the upload_tmpdir, but I might missunderstand something

Reply:
i know. especially since we dont change download_tmpdir, and then as far as i can see dont use it again after yielding.  but when i tried removing it it broke. i cannot explain why

Reply:
ok, I will have a closer look